Update data-retention-policy.rst

wiersgallak opened the pull request:
Adding docs for custom data retention policies.

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
-To set a custom data retention policy:
+To set a global data retention policy:
 
 1. Go to **System Console > Compliance > Data Retention Policy**.
```
```suggestion
1. Go to **System Console > Compliance > Data Retention Policies**.
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
-2. Select a **Message Retention** option. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
-3. Select a **File Retention** option. When a time is specified uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage** at the specified time. The minimum retention period is one day.
-4. Set the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. Must be a 24-hour time stamp in the form HH:MM.
+2. Select edit from the menu presented to the right of the **Global retention policy** table. 
```
```suggestion
2. Select **Edit** from the menu located to the right of the **Global retention policy** table. 
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
-3. Select a **File Retention** option. When a time is specified uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage** at the specified time. The minimum retention period is one day.
-4. Set the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. Must be a 24-hour time stamp in the form HH:MM.
+2. Select edit from the menu presented to the right of the **Global retention policy** table. 
+3. Select a **Channel & direct message retention** option from the dropdown. Specify the number of days or years to keep channel and direct messages. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
```
```suggestion
3. Specify a global retention policy for channel messages and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specifying how long to keep those messages. When a time is specified, messages and file attachments older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
-4. Set the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. Must be a 24-hour time stamp in the form HH:MM.
+2. Select edit from the menu presented to the right of the **Global retention policy** table. 
+3. Select a **Channel & direct message retention** option from the dropdown. Specify the number of days or years to keep channel and direct messages. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
+3. Select a **File retention** option from the dropdown. Specify the number of days or ears to keep files. When a time is specified uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage** at the specified time. The minimum retention  ...
```
```suggestion
3. Specify a global retention policy for files by selecting a **File retention** option from the dropdown, then specifying how long to keep uploaded files. When a time is specified, uploaded files older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage**) at the specified time. The minimum retention period is one day.
```

cwarnermm replied:
We appear to be making a distinction between file attachments on a channel or DM message versus uploaded files. Is this intentional? If so, we may want to clearly emphasize the difference between the two types of "files" and under what conditions each global retention option will work.

mkraft replied:
the number 3 is used for two different bullet.

mkraft replied:
> We appear to be making a distinction between file attachments on a channel or DM message versus uploaded files. Is this intentional? If so, we may want to clearly emphasize the difference between the two types of "files" and under what conditions each global retention option will work.

Every file in the Channels product is attached to a post. The global file policy deletes all files (regardless of whether they're in a direct message, private, or public channel).

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+2. Select edit from the menu presented to the right of the **Global retention policy** table. 
+3. Select a **Channel & direct message retention** option from the dropdown. Specify the number of days or years to keep channel and direct messages. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
+3. Select a **File retention** option from the dropdown. Specify the number of days or ears to keep files. When a time is specified uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage** at the specified time. The minimum retention  ...
+4. Set the start time of the daily scheduled data retention job under the **Policy log** section. Choose a time when fewer people are using your system. 
```
```suggestion
4. Under the **Policy log** section, select **Edit** to specify the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. 
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+4. Set the start time of the daily scheduled data retention job under the **Policy log** section. Choose a time when fewer people are using your system. 
 
-Save the settings and restart your server. Messages and files older than the duration you set will be deleted at the specified server time, if applicable.
+Save the settings. Messages and files older than the duration you set will be deleted at the specified server time, if applicable.
```
```suggestion
Select **Save**. Messages and files older than the duration you set will be deleted at the specified server time, as applicable.
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+Configuring a Custom Data Retention Policy
+-------------------------------------------
+
+To set a custom data retention policy in Mattermost versions 5.38+:
```
```suggestion
To set a custom data retention policy in Mattermost v5.38 and later:
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+-------------------------------------------
 
-To set a custom data retention policy:
+To set a global data retention policy:
```
```suggestion
To set a global data retention policy in Mattermost v5.38 and later:
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+
+To set a custom data retention policy in Mattermost versions 5.38+:
+
+1. Go to **System Console > Compliance > Data Retention Policy**.
```
```suggestion
1. Go to **System Console > Compliance > Data Retention Policies**.
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+To set a custom data retention policy in Mattermost versions 5.38+:
+
+1. Go to **System Console > Compliance > Data Retention Policy**.
+2. Select **Add policy** from the menu presented to the right of the **Custom retention policy** table. 
```
```suggestion
2. Select **Add policy** to the right of the **Custom retention policies** table. 
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+
+1. Go to **System Console > Compliance > Data Retention Policy**.
+2. Select **Add policy** from the menu presented to the right of the **Custom retention policy** table. 
+3. Create a name for your policy. 
```
```suggestion
3. Specify a name for your policy. 
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+1. Go to **System Console > Compliance > Data Retention Policy**.
+2. Select **Add policy** from the menu presented to the right of the **Custom retention policy** table. 
+3. Create a name for your policy. 
+4. Select a **Channel & direct message retention** option from the dropdown. Specify the number of days or years to keep channel and direct messages. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
```
```suggestion
4. Specify a custom retention policy for channel and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specifying how long to keep uploaded files. When a time is specified, messages and file attachments older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+2. Select **Add policy** from the menu presented to the right of the **Custom retention policy** table. 
+3. Create a name for your policy. 
+4. Select a **Channel & direct message retention** option from the dropdown. Specify the number of days or years to keep channel and direct messages. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
+5. Assign teams and channels to this policy by selecting **Add teams** and searching for a specific team or by selecting **Add channels** and searching for a specific channel. All channels will be included in the a policy assigned to a team only. 
```
```suggestion
5. Assign teams and channels to this policy by selecting **Add teams** and searching for a specific team, or by selecting **Add channels** and searching for a specific channel. If only teams are specified, all channels for selected teams will be included in the a policy. 
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -10,18 +10,35 @@ In Mattermost Enterprise E20, you can set a custom policy to manage how long mes
+3. Create a name for your policy. 
+4. Select a **Channel & direct message retention** option from the dropdown. Specify the number of days or years to keep channel and direct messages. When a time is specified, messages, including file attachments, older than the duration you set will be deleted at the specified time. The minimum retention period is one day.
+5. Assign teams and channels to this policy by selecting **Add teams** and searching for a specific team or by selecting **Add channels** and searching for a specific channel. All channels will be included in the a policy assigned to a team only. 
+6. Set the start time of the daily scheduled data retention job under the **Policy log** section. Choose a time when fewer people are using your system. If there is already a time set for global retention policies then this time will apply also to custom data retention policies. 
```
```suggestion
6. Under the **Policy log** section, select **Edit** to specify the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. If a time is already set for a global retention policy, then the same time applies to custom data retention policies. 
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -61,12 +80,6 @@ This usually means another data retention job is in progress. You can verify thi
 
 If no jobs are in progress and the job has stayed ``Pending`` for more than 2 minutes, then you may not have restarted your server after enabling the data retention policy. Restart your server and try again.
```
Is a server restart required when making global or custom data retention policies?

wiersgallak replied:
@mkraft Can you confirm if a restart is still required for making changes to the data retention policies?

mkraft replied:
No it's not necessary anymore.

wiersgallak replied:
Great Thanks!!

cwarnermm commented:
Thank you so much for this great PR, @wiersgallak! 

The introduction contains this line:
"In Mattermost Enterprise E20, you can set a custom policy to manage how long messages and file uploads are kept in Mattermost channels and direct messages." 

We should update this as follows:
"In Mattermost Enterprise E20, you can set a global retention policy as well as custom retention policies to manage how long messages and file uploads are kept in Mattermost channels and Direct Messages.

cwarnermm reviewed (changes requested):
Thank you for this excellent PR for data retention policies. I've provided suggestions and feedback inline.

mkraft reviewed (approved):
At some point we could ask ourselves to what extent we need to documenting seemingly obvious user-experience steps. For example:

> Go to **System Console > Compliance > Data Retention Policy**.

or

> select **Edit** to specify the start time of the daily scheduled data retention job

I just wonder if future documents like this could only contain the non-obvious descriptions, gotchas and implementation details that the UX doesn't outline.

wiersgallak commented:
@cwarnermm I made some editorial updates again along with addressing your feedback.  I felt like we were being a bit redundant and overusing the word "specified".  I also removed the sections that indicated a restart was required.

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -5,23 +5,40 @@ Data Retention Policy (E20)
-4. Set the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. Must be a 24-hour time stamp in the form HH:MM.
+1. Go to **System Console > Compliance > Data Retention Policies**.
+2. Select **Edit** from the menu located to the right of the **Global retention policy** table. 
+3. Specify a global retention policy for channel messages and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specify how long to keep those messages. When a time is specified, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.
```
```suggestion
3. Specify a global retention policy for channel messages and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then set how long to keep those messages. When a time is set, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -5,23 +5,40 @@ Data Retention Policy (E20)
+1. Go to **System Console > Compliance > Data Retention Policies**.
+2. Select **Edit** from the menu located to the right of the **Global retention policy** table. 
+3. Specify a global retention policy for channel messages and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specify how long to keep those messages. When a time is specified, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.
+4. Select a **File retention** option from the dropdown. Specify the number of days or ears to keep files. When a time is specified, uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage**. The minimum retention period is one day. Th ...
```
```suggestion
4. Select a **File retention** option from the dropdown. Set the number of days or ears to keep files. When a time is set, uploaded files which are older than the duration you set will be deleted from your file storage system (either from your local disk or your Amazon S3 service as specified in **System Console > Environment > File Storage**. The minimum retention period is one day. The global file policy deletes all files regardless of whether they're in a direct message, private, or public channel.
```

cwarnermm commented on source/comply/data-retention-policy.rst:
```diff
@@ -5,23 +5,40 @@ Data Retention Policy (E20)
+1. Go to **System Console > Compliance > Data Retention Policies**.
+2. Select **Add policy** to the right of the **Custom retention policies** table. 
+3. Specify a name for your policy. 
+4. Specify a custom retention policy for channel and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then specify how long to keep uploaded files. When a time is specified, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.
```
```suggestion
4. Specify a custom retention policy for channel and Direct Messages by selecting a **Channel & direct message retention** option from the dropdown, then set how long to keep uploaded files. When a time is set, messages and file attachments older than the duration you set will be deleted. The minimum retention period is one day.
```

cwarnermm reviewed (approved):
Thank you for this great docs addition! I've identified a few additional places where we can convert "specified" to "set" for simplicity and to remove unnecessary redundancy.
